### PR TITLE
compute total messages per show (with opt-ins)

### PR DESCRIPTION
**Pull request by IsaackMwenda**
This computes messages per show by reading in a per rqa traced data and updating `messages_per_show_with_optins` dict.  Example file https://drive.google.com/open?id=1JHhbbzj_Y9zdTUkSg7OlbtdJlLgSxm0F

At this point though, I am considering moving all the pending approval scripts https://github.com/AfricasVoices/Project-IMAQAL/pull/122 and https://github.com/AfricasVoices/Project-IMAQAL/pull/123 here (once approved) and rename the file to `generate_engagement_files`. This way, in just one run, we will be able to :
1. Read each RQA traced data.
2. Compute the no. of messages in that episode.
3. Generate a demog map.
4. Use the demog map to compute unique participants per show, lifetime_activations_per_show, and repeat participation  + any other metrics.

### CODE REVIEW

File: generate_demog_maps.py [hunk 1]
```diff
@@ -83,5 +91,12 @@
...
 
         log.info(f'Writing demog map for {len(demog_map.keys())} uids for {rqa_raw_field}')
-        with open(f'{demog_map_json_output_dir}/{rqa_raw_field}_demog_map.json', "w") as f:
+        with open(f'{output_dir}/{rqa_raw_field}_demog_map.json', "w") as f:
             json.dump(demog_map, f)
+
+    log.info(f'Writing messages_per_show csv')
+    # Export the no. of messages per frequency data to a csv
+    with open(f"{output_dir}/messages_per_show.csv", "w") as f:
+        writer = csv.writer(f, lineterminator="\n")
```
Comment (as2388) [inline generate_demog_maps.py]: May be worth a header row

File: generate_demog_maps.py [hunk 2]
```diff
@@ -56,22 +58,28 @@
         PipelineConfiguration.RQA_CODING_PLANS = PipelineConfiguration.FULL_PIPELINE_RQA_CODING_PLANS
 
     rqa_raw_fields =[]
+    messages_per_show = OrderedDict()  # Of radio show index to messages count
```
[the remark this answers is not in the document]
Comment (IsaackMwenda) [reply generate_demog_maps.py]: updated to name thanks

File: generate_demog_maps.py [hunk 3]
```diff
@@ -56,22 +58,28 @@
...
 
-        log.info(f"Generating demog map for {rqa_raw_field}...")
+        log.info(f"Generating demog map and calculating total messages for {rqa_raw_field}...")
         demog_map = dict()
         sys.setrecursionlimit(20000)
         for msg in messages:
-            if msg["uid"] in demog_map.keys():
-                continue
-            if msg.get(rqa_raw_field, "") != "" and msg["consent_withdrawn"] == Codes.FALSE:
+            if msg.get(rqa_raw_field, "") != "" and msg["consent_withdrawn"] == "false":
```
[the remark this answers is not in the document]
Comment (IsaackMwenda) [reply generate_demog_maps.py]: updated thanks

File: generate_demog_maps.py (hunk 1 above)
Comment (IsaackMwenda) [reply generate_demog_maps.py]: added thanks

**Pull request merged**